python/ray/tune/schedulers/median_stopping_rule.py

```diff
@@ -77,24 +83,41 @@ def on_trial_result(self, trial_runner, trial, result):
         value by step `t` is strictly worse than the median of the running
         averages of all completed trials' objectives reported up to step `t`.
         """
-        if self._time_attr not in result or self._metric not in result:
-            return TrialScheduler.CONTINUE
-
         if trial in self._stopped_trials:
             assert not self._hard_stop
-            return TrialScheduler.CONTINUE  # fall back to FIFO
+            # Fall back to FIFO
+            return TrialScheduler.CONTINUE
 
         time = result[self._time_attr]
         self._results[trial].append(result)
-        median_result = self._get_median_result(time)
+
+        if time < self._grace_period:
+            return TrialScheduler.CONTINUE
+
+        trials = self._trials_beyond_time(time)
+        trials.remove(trial)
```

Comment:
is this needed?

Reply:
It's more consistent with the Vizier paper's alg, i think they don't include it in the median computation. It also makes sense to me that if someone wants `k` samples to be used to calculate the median that that shouldn't include the current trial. Aside from that, whether or not it's really necessary is unclear to me.